src/meta/meta.hpp

```diff
@@ -2054,6 +2071,16 @@ static bool compare(char, const void *lhs, const void *rhs) {
     return lhs == rhs;
 }
 
+template <typename> struct is_template : std::false_type {
+    using type = void;
+    using inner_type = void;
+};
+
+template <template <typename T> class Tmpl, typename T>
+struct is_template<Tmpl<T>> : std::true_type {
```

Comment:
did you test that this worked with vector, since vector is actually std::vector<T, AllocatorT<T>>

Reply:
Fixed it so it now has multiple types, and you can introspect each of them.
It likely doesn't work for all templates, but for things like std::vector and std::shared_ptr that has multiple but can specialize with one it should work fine